python/src/aac/plugins/rest_api/aac_rest_api_impl.py

```diff
@@ -0,0 +1,29 @@
+"""AaC Plugin implementation module for the aac-rest-api plugin."""
+# NOTE: It is safe to edit this file.
+# This file is only initially generated by the aac gen-plugin, and it won't be overwritten if the file already exists.
+
+from typing import Optional
+import uvicorn
+
+from aac.plugins.plugin_execution import PluginExecutionResult, plugin_result
+from aac.plugins.rest_api.aac_rest_app import app
+
+plugin_name = "aac-rest-api"
+
+
+def rest_api(host: Optional[str], port: Optional[int]) -> PluginExecutionResult:
+    """
+    Start a RESTful interface for interacting with and managing AaC.
+
+    Args:
+        host (Optional[str]): Set the hostname of the service. Useful for operating behind proxies.
+        port (Optional[int]): The port to bing the RESTful service to.
+    """
+    with plugin_result(plugin_name, _start_restful_service, host, port) as result:
+        return result
+
+
+def _start_restful_service(host: Optional[str], port: Optional[int]) -> None:
+    """Start the RESTful interface service."""
+    uvicorn.run(app, host=host or "0.0.0.0", port=port or 8000)
```

Comment:
Correct, uvicorn does block until killed. It's the best case since it allows the active context to persist between REST calls -- something that's lacking from the CLI. 

We don't want to link the LSP and the REST API -- the implementation would be too difficult and complex, but if we let the REST server alter filesystem files, then the LSP editor should pick up the changes and effectively synchronize to the new content. The only downside for letting the REST server stomp files is that it will cause issues with the VSCode editor and unsaved changes, which the user will ultimately have learn to deal with. Think of the REST API as an alternative to the LSP -- the user will have to use one or the other, but they can switch between modes.

Reply:
For the first item, we decided on updating the VSCode extension to start the REST server after which we will start the LSP server via a call on the REST API (which will require completion of #351).